src/App.tsx

```diff
@@ -1,223 +1,169 @@
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotification } from
+  './components/ErrorNotification/ErrorNotification';
+import { Todo } from './types/Todo';
+import { addTodo, getTodos, removeTodo } from './api/todos';
+import { FilterType } from './types/FiltersType';
+import { getFilteredTodos } from './components/helpers/getFilteredTodos';
+import { TodoItem } from './components/TodoItem/TodoItem';
+
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedFilter, setCompletedFilter] = useState(FilterType.all);
+  const [errorText, setErrorText] = useState('');
+
+  const [newTitle, setNewTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletingTodosIds, setDeletingTodosIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => setErrorText('Unable to load a todos'));
+    }
+  }, []);
+
+  const addNewTodo = useCallback((event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!newTitle) {
+      setErrorText('Title can\'t be empty');
+
+      return;
+    }
+
+    const newTodoAdding = async () => {
+      setIsAdding(true);
+
+      if (user) {
+        setTempTodo({
+          id: 0,
+          userId: user?.id,
+          title: newTitle,
+          completed: false,
+        });
+
+        try {
+          const newTodo = await addTodo({
+            userId: user.id,
+            title: newTitle,
+            completed: false,
+          });
+
+          setTempTodo(null);
+
+          setTodos(currentTodos => [...currentTodos, newTodo]);
+
+          setNewTitle('');
+        } catch (e) {
+          setErrorText('Unable to add a todo');
+        } finally {
+          setIsAdding(false);
+        }
+      }
+    };
+
+    newTodoAdding();
+  }, [newTitle]);
+
+  const deleteTodo = useCallback(async (todoId: number) => {
+    setDeletingTodosIds(prev => [...prev, todoId]);
+    try {
+      await removeTodo(todoId);
+
+      setTodos(currentTodos => currentTodos.filter(
+        todo => todo.id !== todoId,
+      ));
+    } catch (e) {
+      setErrorText('Unable to delete a todo');
+    } finally {
+      setDeletingTodosIds(prev => {
+        return prev.filter(deletingId => deletingId !== todoId);
+      });
+    }
   }, []);
 
+  const deleteAllComplitedTodos = useCallback(() => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        deleteTodo(todo.id);
+      }
+    });
+  }, [todos]);
```

Comment:
Add ids to DeletingTodosIds and remove them after deleting. Promise.all could help u